include/swift/Reflection/TypeRefBuilder.h

```diff
@@ -572,9 +570,19 @@ class TypeRefBuilder {
         args.end() - argsIndex - numGenericArgs, args.end() - argsIndex);
 
     const BoundGenericTypeRef *parent = nullptr;
-    if (node->hasChildren())
-     parent = createBoundGenericTypeReconstructingParent(
-        node->getFirstChild(), decl, --shapeIndex, args, argsIndex + numGenericArgs);
+    if (node->hasChildren()) {
+      // Skip over nodes that are not of type class, enum or struct
+      auto parentNode = node->getFirstChild();
+      while (parentNode->getKind() != Node::Kind::Class &&
```

Comment:
There could be a null pointer deref here.

Reply:
Let me know if this looks reasonable

Reply:
LGTM :)